Plugins/LoggerPlugin/LoggerPlugin.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import PackagePlugin
+import Foundation
+
+@main
+struct LoggerPlugin: BuildToolPlugin {
+  func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+    let outputDirectory = context.pluginWorkDirectory.appending("GeneratedSources")
+    guard let target = target as? SourceModuleTarget else {
+      Diagnostics.error("Attempted to use `LoggerPlugin` on an unsupported module target")
+      return []
+    }
+    
+    try FileManager.default.createDirectory(atPath: outputDirectory.string, withIntermediateDirectories: true)
+    let source = """
+    import Foundation
+    import os.log
+    
+    public extension Logger {
+      static var module: Logger {
+        .init(subsystem: "\\(Bundle.main.bundleIdentifier ?? "com.brave.ios")", category: "\(target.moduleName.lowercased())")
```

Comment:
There a reason to lowercasing the category here? For multiple word targets like ("BraveNews" → "bravenews") lowercasing them would make the categories less readable